framework/base/src/types/managed/wrapped/cached_managed_buffer.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{
+    api::{ErrorApiImpl, ManagedTypeApi},
+    types::{ManagedBuffer, StaticBufferRef},
+};
+
+fn load_static_cache<M>(managed_buffer: &ManagedBuffer<M>) -> StaticBufferRef<M>
+where
+    M: ManagedTypeApi,
+{
+    StaticBufferRef::try_new_from_copy_bytes(managed_buffer.len(), |dest_slice| {
+        let _ = managed_buffer.load_slice(0, dest_slice);
+    })
+    .unwrap_or_else(|| M::error_api_impl().signal_error(b"Static cache is in use"))
```

Comment:
Slightly misleading error message - this error can also occur when the static buffer is too small for loading the managed buffer (the limit is 10000 bytes at the moment) - see [try_lock_with_copy_bytes](https://github.com/ElrondNetwork/elrond-wasm-rs/blob/7cf42c8e03c8b0543a4ca51ebf028b5416166d9d/framework/base/src/types/static_buffer/lockable_static_buffer.rs#L39)

Reply:
```suggestion
    .unwrap_or_else(|| M::error_api_impl().signal_error(b"static cache too small or already in use"))
```